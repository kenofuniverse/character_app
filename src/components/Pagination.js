import React from 'react'
import PropTypes from 'prop-types'
import { Pagination } from 'react-bootstrap'
import _ from 'lodash';

const generatePageNumbers = (totalPages, currentPage) => {
  const limit = 5;
  const offset = 2;

  if (currentPage <= offset + 1) {
    return _.range(1, limit > totalPages ? totalPages + 1 : limit + 1);
  }
  if (currentPage >= totalPages - offset) {
    return _.range(limit > totalPages ? 1 : totalPages - limit + 1, totalPages + 1);
  }
  return _.range(currentPage - offset, currentPage + offset + 1);
}

const PaginationButons = ({current, total, onClickItem}) => (
  <Pagination>
    <Pagination.Prev disabled={current <= 1} onClick={() => onClickItem(current - 1)} />
    { generatePageNumbers(total, current).map((page, index) => (
        <Pagination.Item key={index} active={page === current} onClick={() => onClickItem(page)}>{page}</Pagination.Item>
      ))
    }
    <Pagination.Next disabled={current >= total} onClick={() => onClickItem(current + 1)} />
  </Pagination>
)

PaginationButons.propTypes = {
  current: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired,
  onClickItem: PropTypes.func.isRequired
}

export default PaginationButons
