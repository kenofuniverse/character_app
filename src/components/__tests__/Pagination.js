import React from 'react'
import renderer from 'react-test-renderer'
import Pagination from '../Pagination'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Pagination current={1} total={10} onClickItem={() => {}} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
