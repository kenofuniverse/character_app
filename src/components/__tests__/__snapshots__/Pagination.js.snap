// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="pagination"
>
  <li
    className="page-item disabled"
  >
    <span
      className="page-link"
      disabled={true}
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      >
        ‹
      </span>
      <span
        className="sr-only"
      >
        Previous
      </span>
    </span>
  </li>
  <li
    className="page-item active"
  >
    <span
      className="page-link"
      disabled={false}
      onClick={[Function]}
    >
      1
      <span
        className="sr-only"
      >
        (current)
      </span>
    </span>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      2
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      3
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      4
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      5
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      >
        ›
      </span>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </li>
</ul>
`;
